[nnc] ported some more ops + added vectors to argvalue

Chillee opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#56766 [nnc] ported some more ops + added vectors to argvalue**

Differential Revision: [D28118331](https://our.internmc.facebook.com/intern/diff/D28118331)

Chillee committed:
[nnc] ported some more ops + added vectors to argvalue

[ghstack-poisoned]

Chillee committed:
Update on "[nnc] ported some more ops + added vectors to argvalue"




[ghstack-poisoned]

Chillee committed:
Update on "[nnc] ported some more ops + added vectors to argvalue"




[ghstack-poisoned]

Chillee committed:
Update on "[nnc] ported some more ops + added vectors to argvalue"




[ghstack-poisoned]

Chillee committed:
Update on "[nnc] ported some more ops + added vectors to argvalue"




[ghstack-poisoned]

Chillee committed:
Update on "[nnc] ported some more ops + added vectors to argvalue"




[ghstack-poisoned]

Chillee committed:
Update on "[nnc] ported some more ops + added vectors to argvalue"




[ghstack-poisoned]

Chillee committed:
Update on "[nnc] ported some more ops + added vectors to argvalue"




[ghstack-poisoned]

Chillee committed:
Update on "[nnc] ported some more ops + added vectors to argvalue"




[ghstack-poisoned]

Chillee committed:
Update on "[nnc] ported some more ops + added vectors to argvalue"




[ghstack-poisoned]

Chillee committed:
Update on "[nnc] ported some more ops + added vectors to argvalue"




[ghstack-poisoned]

Chillee committed:
Update on "[nnc] ported some more ops + added vectors to argvalue"




[ghstack-poisoned]

Chillee committed:
Update on "[nnc] ported some more ops + added vectors to argvalue"




[ghstack-poisoned]

ZolotukhinM commented on torch/csrc/jit/tensorexpr/kernel.cpp:
```diff
@@ -976,15 +990,167 @@ Tensor* computeFourOperand(
+  std::vector<BufHandle> nonEmptyInputs;
+  for (auto buf : bufList) {
+    bufInputs.push_back(buf);
+    assert(buf.node()->dims().size() > 0);
```
s/assert/TORCH_INTERNAL_ASSERT/

ZolotukhinM commented on torch/csrc/jit/tensorexpr/kernel.cpp:
```diff
@@ -976,15 +990,167 @@ Tensor* computeFourOperand(
+    }
+    nonEmptyInputs.push_back(buf);
+  }
+  auto maybe_dtype = bufInputs[0].dtype().scalar_type();
```
The "maybe" prefix usually indicates that the value is optional. Here it doesn't seem such.

ZolotukhinM commented on torch/csrc/jit/tensorexpr/kernel.h:
```diff
@@ -25,14 +25,24 @@ inline std::vector<int64_t> bufferSizes(const T& t) {
 }
 using ArgNone = c10::monostate;
 using BufList = std::vector<tensorexpr::BufHandle>;
+using IntList = std::vector<int64_t>;
```
`IntList` doesn't seem to be used anywhere.

Chillee replied:
This is used in some other lowerings - I just added all the lists here.

ZolotukhinM reviewed (approved):
Looks good to me!

Chillee committed:
Update on "[nnc] ported some more ops + added vectors to argvalue"




[ghstack-poisoned]

Commit:
Update on "[nnc] ported some more ops + added vectors to argvalue"


Differential Revision: [D28118331](https://our.internmc.facebook.com/intern/diff/D28118331)

[ghstack-poisoned]

Commit:
Update on "[nnc] ported some more ops + added vectors to argvalue"


Differential Revision: [D28118331](https://our.internmc.facebook.com/intern/diff/D28118331)

[ghstack-poisoned]

Commit:
Update on "[nnc] ported some more ops + added vectors to argvalue"


Differential Revision: [D28118331](https://our.internmc.facebook.com/intern/diff/D28118331)

[ghstack-poisoned]
